lib/tests/misc.nix

```
@@ -564,6 +564,115 @@ runTests {
     };
   };
 
+  # code from the example
+  testCollect'Example = let
```

Maybe 
```suggestion
  testCollectPrimeExample = let
```

Since your test seems to be the first one that tests a function ending in "prime", following test contributions might the naming convention.

These are a little awkward to type:

`mapAttrs'Example`
`overrideScope'Example`

---

I think this is a PrimeExample of an unnecessary rule that may lead to more discomfort rather than less.

To be fair, I wouldn't actually call it a _prime_ example, but I couldn't resist the pun, and it actually makes a point!
However, test names are a bit inconsequential.